### ISSUE TITLE
histeq duplicate def

### ISSUE THREAD

**Issue by bjarthur** [timholy/Images.jl#517]
@mronian, do you know about this?

```
julia> using Images
INFO: Recompiling stale cache file /Users/arthurb/.julia/lib/v0.4/Images.ji for module Images.
WARNING: New definition 
    histeq(AbstractArray{#T<:Union{ColorTypes.Gray, Number}, N<:Any}, Int64, #T<:Union{ColorTypes.Gray, Number}, #T<:Union{ColorTypes.Gray, Number}) at /Users/arthurb/.julia/v0.4/Images/src/algorithms.jl:1328
is ambiguous with: 
    histeq(Images.AbstractImage, Any, Any, Any) at /Users/arthurb/.julia/v0.4/Images/src/algorithms.jl:1285.
To fix, define 
    histeq(Images.AbstractImage{_<:Any, N<:Any}, Int64, _<:Any, _<:Any)
before the new definition.
```

**Comment by timholy** [comment]
This is fixed in #516. If @mronian has only been testing Julia 0.5, ambiguities are hidden until you try to call them, so they are harder to notice. (I often run `Base.Test.detect_ambiguities(Images, Base)` when developing.)

See also https://github.com/JuliaLang/julia/issues/17444.

**Comment by mronian** [comment]
@bjarthur Thanks for pointing out this bug :)
